src/components/ModalInput/ModalInput.tsx

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Text, Button, Input, InputProps, Flex, Link } from '@pancakeswap-libs/uikit'
+import useI18n from '../../hooks/useI18n'
+
+interface ModalInputProps {
+  max: string
```

Comment:
This was previously `number | string` - do we know why? Having the potential for multiple types caused some problems with me trying to convert the string to a float and then round, later on in the component.